src/Azure.Functions.Cli/Actions/LocalActions/CreateFunctionAction.cs

```diff
@@ -21,18 +22,23 @@ namespace Azure.Functions.Cli.Actions.LocalActions
     [Action(Name = "create", Context = Context.Function, HelpText = "Create a new function from a template.")]
     internal class CreateFunctionAction : BaseAction
     {
-        private readonly ITemplatesManager _templatesManager;
+        private ITemplatesManager _templatesManager;
         private readonly ISecretsManager _secretsManager;
 
         public string Language { get; set; }
         public string TemplateName { get; set; }
         public string FunctionName { get; set; }
         public bool Csx { get; set; }
+        public HttpAuthorizationLevel? AuthorizationLevel { get; set; }
```

Comment:
> public HttpAuthorizationLevel? AuthorizationLevel { get; set; } [](start = 7, length = 64)

Is there a scenario where this can/should be null ?

Reply:
The main reason for us to have a null here is because the `authlevel` does not apply to other triggers. The `func new` and its parameters until now were applicable to all the trigger. So if the user decides to use this for other templates that does not accept the authLevel parameter, I wanted to provide an error message. Without a null value i would have no way to differentiate between no value passed vs default value without making it nullable.

Reply:
I was thinking along the lines of making default to be "None" and if for other triggers , if it's anything other than "None" , an error could be displayed. But then this means httpTrigger should have some code changes where default cannot be "None". 

I don't really have much of a preference, either ways.

Reply:
Ahh i thought you were saying use one of  the values as default instead. I used "unassigned" as a default value to signal null equivalent. But then I found out that i would need to add custom parsing logic and validation logic. Adding nullable allows me use default validation from the framework. Also now I am using the Enum from webjobs sdk, So i can't add a default anyway.

Reply:
appreciate the feedback :)